weather_mv/loader_pipeline/sinks.py

```diff
@@ -406,7 +409,19 @@ def open_dataset(uri: str,
     """Open the dataset at 'uri' and return a xarray.Dataset."""
     try:
         if is_zarr:
+            start_date = end_date = None
+            if 'start_date' in open_dataset_kwargs:
+                start_date = open_dataset_kwargs['start_date']
+                if open_dataset_kwargs is not None:
+                    open_dataset_kwargs.pop('start_date', None)
```

Comment:
```suggestion
                start_date = open_dataset_kwargs.pop('start_date', None)                    
```

Reply:
Hi @alxmrs, it's important to ensure that a dictionary is not **empty** before using the `pop` operation directly on it. Therefore, we have to must add a condition like `if open_dataset_kwargs is not None:` before using `pop` operation.

Despite making code updates, I aimed to keep the code as minimal as possible.